subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -1270,8 +1228,11 @@ class TestSuitesGroovyDSLDependenciesIntegrationTest extends AbstractIntegration
             }
         """
 
-        expect:
-        succeeds 'checkConfiguration'
+        when:
+        fails 'checkConfiguration'
+
+        then:
+        failureHasCause("class java.lang.String cannot be cast to class org.gradle.api.artifacts.Dependency")
```

Comment:
This is a pretty crappy error.  Is there something we could do better here? e.g., suggest that you use `dependencyFactory` or another method?

Reply:
Done.